File: dotnet/CudaRS.Native/SdkPipelineSpec.cs
using System;
using System.Runtime.InteropServices;

namespace CudaRS.Native;

[StructLayout(LayoutKind.Sequential)]
public struct SdkPipelineSpec
{
    public IntPtr IdPtr;
    public nuint IdLen;
    public SdkPipelineKind Kind;
    public IntPtr ConfigJsonPtr;
    public nuint ConfigJsonLen;
}

File: cudars-ffi/src/sdk/sdk_pipeline_spec.rs
use libc::c_char;

use cudars_core::PipelineKind;

#[repr(C)]
pub struct SdkPipelineSpec {
    pub id_ptr: *const c_char,
    pub id_len: usize,
    pub kind: PipelineKind,
    pub config_json_ptr: *const c_char,
    pub config_json_len: usize,
}
